### DISCUSSION TITLE
Is there a reason NOT to use a Provider?

### DISCUSSION THREAD

**Discussion by megafinz** [pmndrs/jotai#1709] [Q&A] [answered]
Yello!

I've been trying to wrap my head around using Jotai in SSR scenario with NextJS for a couple of days now. As far as I can understand, it is recommended to wrap the root of the App in a Provider element so when the App is recreated on each server request, the atom store will be recreated with the Provider as well, otherwise atoms will use a provider-less store that is global and is retained between different requests, which can possibly lead to leaking information from unrelated requests. This wasn't very clear from documentation so it took me some time to come to this conclusion. Am I correct here?

Now, the question is: aside from a minor convenience, is there a case where NOT wrapping your App in a "root" Provider is better than wrapping it? It seems to me that Provider is essential for SSR, so making the Provider required in both SSR and CSR scenarios will reduce the mental effort required to setup Jotai in all cases. You don't need to think about whether you need a Provider or not, just use it all the time and that's it.

Does that make sense?

**Comment by dai-shi** [comment, marked as answer]
> Am I correct here?

Yes, absolutely. Can you help improving docs?
Feel free to send a PR to fix https://github.com/pmndrs/jotai/blob/main/docs/guides/nextjs.mdx.

> is there a case where NOT wrapping your App in a "root" Provider is better than wrapping it?

Some people **really** like it. Quite a few people got confused when they forgot to put Provider (when we didn't have the provider-less support), and we just need to answer that they forgot Provider to such questions.

> You don't need to think about whether you need a Provider or not, just use it all the time and that's it.

Your point makes certain sense. After all, it's trade-off. Poimandres state libs are basically unopinionated, so it's a design decision.

**Reply by megafinz** [reply]
> Can you help improving docs?

Sure thing: https://github.com/pmndrs/jotai/pull/1710

> Some people really like it. Quite a few people got confused when they forgot to put Provider (when we didn't have the provider-less support), and we just need to answer that they forgot Provider to such questions.

Hmm, I'd expect that React users are accustomed to using providers, maybe the error message was not very helpful? For example if you forget to use a provider with Apollo Client (and don't pass the client instance directly into query/mutation hooks) it will give a pretty clear explanation of what you must do to fix the problem.

> After all, it's trade-off. Poimandres state libs are basically unopinionated, so it's a design decision.

Makes sense.
